provider/provider.go

```diff
@@ -503,6 +586,19 @@ func (s *SweepingProvider) reprovideTimeForPrefix(prefix bitstr.Key) time.Durati
 	return time.Duration(int64(s.reprovideInterval) * val / maxInt)
 }
 
+// nextPrefixToReprovideNoLock returns the next prefix to reprovide, based on
+// the current time offset.
+func (s *SweepingProvider) nextPrefixToReprovideNoLock(offset time.Duration) (bitstr.Key, error) {
+	maxVal := 1 << maxPrefixSize
+	intPrefix := uint64(float64(offset) / float64(s.reprovideInterval) * float64(maxVal))
```

Comment:
Could you avoid floating point arithmetic if you did the multiplication first:
```
intPrefix := offset * maxVal / uint64(s.reprovideInterval)
```
Should probably do the multiplication first anyway to minimize rounding error.

Reply:
In the tests, I faced integer overflow issues when running the multiplication first. Dividing first, and then multiplying avoids the risk of integer overflow.

I'll add a comment (https://github.com/libp2p/go-libp2p-kad-dht/pull/1170/commits/3e1dcd88eea2022c9fd6ae1e01a49b54277baa1e)